test/fixtures/rules/std_rule_cases/L003.yml

```diff
@@ -323,6 +323,7 @@ test_pass_indent_indent_bigquery:
 
 
 test_jinja_indent_templated_table_name_a:
+  # TODO partial fix - this requires some more attention
```

Comment:
I couldn't get that working

Reply:
Please go ahead and commit the change. I can help look into it. These YAML tests sometimes behave strangely because they don't run precisely the normal linter code. In the long run, it'd be cool to clean that up.

Reply:
Actually I'm not entirely sure which line you mean by line 11. Do you mean this:

```
{{ product }} -- noqa: disable=L003
```

or this:

```
{% if not loop.last -%} UNION ALL {%- endif %} -- noqa: disable=L003 
```
?

I tried both but neither of them seemed to help.

Reply:
@barrywhart bumping this comment in case you missed it by accident ^

Reply:
`-- noqa: disable=L003` means disable L003 from this line and every subsequent line going forward (until either end of file or you see an `enable` equivalent).

That's probably overkill (you actually WANT the rule to run on all lines - except the troublesome ones we know we have an issue fixing).

`-- noqa: L003` means just disable L003 from this line, then carry on checking it again from future lines. I think that is the option your want.

Reply:
You're correct: I had the wrong `noqa` syntax!

Reply:
@juhoautio: Please add it to this line: `{% if not loop.last -%} UNION ALL {%- endif %}`, then tag me once you push it. I will look and see why `noqa` is not working.

Reply:
Pushed it https://github.com/sqlfluff/sqlfluff/pull/2624/commits/fe66a93e20a46d2b640e0b9524f94f1ff88774dc

Reply:
See my comment elsewhere -- we need to change how `rules.py` is checking for remaining lint errors, after the fix.

Reply:
If you prefer, I have a suggestion similar to your idea about `fail_after_fix`. I suggest we add a property called `failures_after_fix`, which specifically lists the failures remaining after fixes are applied, e.g.:
```
failures_after_fix:
- code: L003
  description: 'Indentation not consistent with line #9'
  line_no: 11
  line_pos: 1
- code: L003
  description: 'Indentation not consistent with line #19'
  line_no: 11
  line_pos: 1
```

I generated the YAML above with the following code:
```
print(yaml.dump([e.get_info_dict() for e in lerrs]))
```

Thoughts, @juhoautio and @tunetheweb?

Reply:
Nice, that would satisfy the need for asserting errors that I had in mind. Looks good to me!

In the same fashion this could be an optional extension for the cases where there's only `fail_str`, if one wants to add stronger assertions. In that case it could mean for example adding:
```
failures:
- code: L003
  description: 'Indentation not consistent with line #9'
  line_no: 11
  line_pos: 1
```

> I generated the YAML above with the following code

Wow! Seems like it would be quite easy to also print that out in case the detected failures don't match with the expectation.

Reply:
@barrywhart @tunetheweb I can try to implement the strategy that's proposed in https://github.com/sqlfluff/sqlfluff/pull/2624/files#r807452356 if you'd want me to. On the other hand I'm wondering if test cases like this should be moved to be written in python. Could you suggest where I should place such tests? We could then even make the assertion error just say that hey, your test has still failures after some fixes happened, and in this type of case it's best to write the test in python because the yaml framework doesn't cover complicated cases like that. Thoughts?

Reply:
I prefer to extend the testing framework. It seems like a simple enough extension, and it avoids potentially moving and translating tests from one format to another as the rule changes over time.

Reply:
How many tests would we be moving to Python?

Reply:
FYI @juhoautio here’s where the Python test cases for rules live: https://github.com/sqlfluff/sqlfluff/tree/main/test/rules

Reply:
> How many tests would we be moving to Python?

It should be only the ones that I marked as TODO in the diff of this PR, so 4 affected files and just 8 test cases in total. My first PR covers 2 of them and I can work on the rest in the same fashion.